spark/dl/src/test/scala/com/intel/analytics/bigdl/dataset/MiniBatchSpec.scala

```diff
@@ -109,4 +109,34 @@ class MiniBatchSpec extends FlatSpec with Matchers {
     target should be (expectedTarget)
   }
 
+  "SparseTensorMiniBatch with different TensorTypes" should "return right result" in {
+    val a1 = Tensor.sparse(Tensor[Double](4).range(1, 4, 1)).asInstanceOf[Tensor[Float]]
```

Comment:
should we hide this piece of code into the TensorSample.apply()?

Reply:
It looks great!

Reply:
Since __Array[Tensor[T]]__ is __not__ a sub type of __Array[Tensor[_]]__,  it will cause compile error if changing apply method directly.
So, I'm trying to add a _TensorSample.create_ method instead.